### PR TITLE
Jocelyn Gonzalez -- Carets

### PR DESCRIPTION
**Pull request by jocegonz**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| How did you go about exploring the Edamam API, how did you try querying the API? | I read the documentation from Edamam and query results via Postman and my Chrome browser. |
| Describe your API Wrapper.  How did you decide on the methods you created? | My API wrapper has two methods that dictate what information needs to be accessed from the API's database. One method searches for a list of food matches, the other is to show a specific recipe from the collection of search results. I determined my methods based on what the API's documentation gave me access to. |
| Describe an edge case or failure case test you wrote for your API Wrapper. | I tested if the wrapper would return an empty array for a bad uri. |
| Explain how VCR aids in testing an API. | VCRs record responses from the API without having to pull information from the API every time just to test. This saves money and makes you less dependent on the API for tests in case the API went down. |
| What is the Heroku URL of your deployed application? | https://munchsquad.herokuapp.com/ |

### CODE REVIEW

File: lib/edamam_api_wrapper.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          recipe["image"],
+          uri: recipe["uri"],
+          url: recipe["url"],
+          ingredients: recipe["ingredientLines"],
+          nutrition: recipe["digest"]
+        )
+      end
+      return view_results
+    else
+      return []
```
Comment (CheezItMan) [inline lib/edamam_api_wrapper.rb:56]: Why would this return an empty array instead of `nil`?  This is for an individual recipe not a collection.

File: test/controllers/recipes_controller_test.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  it "can view the search page" do
+    VCR.use_cassette("index") do
+      get recipes_path(EdamamApiWrapper.search("chocolate"))
+      must_respond_with :success
+    end
+  end
+
+  it "can view the show page" do
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:18]: What about negative case tests?

File: test/lib/edamam_api_wrapper_test.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+require 'test_helper'
```
Comment (CheezItMan) [inline test/lib/edamam_api_wrapper_test.rb:1]: What about tests for your `Recipe` class?

File: app/views/recipes/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,42 @@
+
+
+<article class = "show row">
+
+    <% @recipe.each do |r| %>
+    <section class="ingredients small-12 large-6 column">
+      <h2><%= r.label %></h2>
+      <p><em>Originally found in <%= link_to r.url %></em></p>
```
Comment (CheezItMan) [inline app/views/recipes/show.html.erb:8]: This link is broken!

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
| **Core Requirements** |   |
| Git hygiene | Good number of commits, but some of your commit messages are a bit... odd.  Three `httparty` commit? |
| Comprehension questions	| Check |
| **General** |  |
| Rails fundamentals (RESTful routing, use of named paths) | Check |
| Semantic HTML |  Good semantic HTML |
| Errors are reported to the user | Check |
| API Wrapper to handle the API requests | Check |
| Controller testing | Check, but no negative tests. |
| Lib testing | Missing a test for your recipe class. |
| Search Functionality | Works! |
| List Functionality | Works! |
| Show individual item functionality (link to original recipe opens in new tab) | **BROKEN** |
| **Styling** |  |
| Foundation Styling for responsive layout | Nice attractive layout, you obviously spent a lot of time working on it. |
| List View shows 10 items at a time/pagination | `hi!!! pagination isn't working! what's up with that???`  <-- That's my line! |
| The app is styled to create an attractive user interface | Nicely done |
| **API Features** | |
| The App attributes Edaman |Check  |
| The VCR casettes do not contain the API key | Check  |
| **External Resources** |  |
| Link to deployed app on Heroku | Check |
|  **Overall** | Nicely done, you had trouble with pagination and didn't cover negative cases in testing, but overall pretty good. |
